custom_components/tuya_local/devices/fresco_hydrateultra_petfountain_alt.yaml

```diff
@@ -0,0 +1,233 @@
+    name: Water level
+    icon: "mdi:water"
+    category: diagnostic
+    dps:
+      - id: 9
+        type: string
+        unit: "%"
+        name: sensor
+        mapping:
+          - dps_val: "1"
+            value: 25
+          - dps_val: "2"
+            value: 50
+          - dps_val: "3"
+            value: 75
+          - dps_val: "4"
+            value: 100
+
+  - entity: light
+    translation_key: indicator
+    category: config
+    dps:
+      - id: 12
+        name: brightness
+        type: integer
+        range:
+          min: 1
+          max: 10
+
+  - entity: sensor
+    name: Water quality
+    icon: "mdi:water"
+    category: diagnostic
+    dps:
+      - id: 13
+        type: integer
+        name: sensor
+        unit: TDS
+
+  - entity: button
+    name: UV sterilization
+    icon: "mdi:lightbulb-fluorescent-tube"
+    category: config
+    dps:
+      - id: 14
+        type: boolean
+        name: button
+        optional: true
+
+  - entity: sensor
+    name: UV runtime
+    class: duration
+    category: diagnostic
+    dps:
+      - id: 15
+        type: integer
+        name: sensor
+        unit: s
+
+  - entity: sensor
+    name: Last Water change
+    icon: "mdi:water-circle"
+    category: diagnostic
+    class: duration
+    dps:
+      - id: 101
+        type: integer
+        name: sensor
+        range:
+          min: 0
+          max: 7200
+        unit: min
+
+  - entity: sensor
+    name: Water health
+    icon: "mdi:water-circle"
+    category: diagnostic
+    class: duration
+    dps:
+      - id: 101
+        type: integer
+        name: sensor
+        range:
+          min: 0
+          max: 7200
+        mapping:
+          - target_range:
+              min: 100
+              max: 0
+        precision: 1
+        unit: "%"
+
+  - entity: button
+    name: Reset Water health
+    icon: "mdi:sync"
+    category: config
+    dps:
+      - id: 101
+        type: integer
+        name: button
+        mapping:
+          - dps_val: 0
+            value: true
+
+  - entity: sensor
+    name: Last Filter change
+    icon: "mdi:air-filter"
+    category: diagnostic
+    class: duration
+    dps:
+      - id: 102
+        type: integer
+        name: sensor
+        range:
+          min: 0
+          max: 43200
+        unit: "m"
+
+  - entity: sensor
+    name: Filter health
+    icon: "mdi:air-filter"
+    category: diagnostic
+    class: duration
+    dps:
+      - id: 102
+        type: integer
+        name: sensor
+        range:
+          min: 0
+          max: 43200
+        mapping:
+          - target_range:
+              min: 100
+              max: 0
+        precision: 1
+        unit: "%"
+
+
+  - entity: button
+    name: Reset Filter health
+    icon: "mdi:sync"
+    category: config
+    dps:
+      - id: 102
+        type: integer
+        name: button
+        mapping:
+          - dps_val: 0
+            value: true
+
+  - entity: sensor
+    name: Last pump maintenance
```

Comment:
As above, invert and rename to "Pump maintenance due", to eliminate the need for a duplicate "percentage" sensor to give a better idea when it comes due.

Reply:
changed as above.